cmd/crank/beta/trace/internal/resource/client_test.go

```diff
@@ -0,0 +1,292 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/claim"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/composite"
+)
+
+type xrcOpt func(c *claim.Unstructured)
+
+func withXRCRef(ref *v1.ObjectReference) xrcOpt {
+	return func(c *claim.Unstructured) {
+		c.SetResourceReference(ref)
+	}
+}
+
+func withXRCSecretRef(ref *xpv1.LocalSecretReference) xrcOpt {
+	return func(c *claim.Unstructured) {
+		c.SetWriteConnectionSecretToReference(ref)
+	}
+}
+
+func buildXRC(opts ...xrcOpt) *unstructured.Unstructured {
+	c := claim.New()
+	for _, f := range opts {
+		f(c)
+	}
+	return &c.Unstructured
+}
+
+type xrOpt func(c *composite.Unstructured)
+
+func withXRRefs(refs ...v1.ObjectReference) xrOpt {
+	return func(c *composite.Unstructured) {
+		c.SetResourceReferences(refs)
+	}
+}
+
+func withXRSecretRef(ref *xpv1.SecretReference) xrOpt {
+	return func(c *composite.Unstructured) {
+		c.SetWriteConnectionSecretToReference(ref)
+	}
+}
+
+func withNamespace(ns string, res *unstructured.Unstructured) *unstructured.Unstructured {
+	res.SetNamespace(ns)
+	return res
+}
+
+func buildXR(opts ...xrOpt) *unstructured.Unstructured {
+	c := composite.New()
+	for _, f := range opts {
+		f(c)
+	}
+	return &c.Unstructured
+}
+
+func TestGetResourceChildrenRefs(t *testing.T) {
+	type args struct {
+		resource   *Resource
+		witSecrets bool
+	}
+	type want struct {
+		refs []v1.ObjectReference
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"XRCWithChildrenMR": {
+			reason: "Should return the XR child for an XRC.",
```

Comment:
the test case name mentions `MR` but the test case content looks to mention XRs instead - is this test case testing what you expect it to?